tutorials/first-steps-6.md

```diff
@@ -0,0 +1,316 @@
+	}
+]
+{% endhighlight %}
+
+Flows have a unique `UUID` per capture but also have a `TrackingID` which is the same across multiple captures.
+
+{% highlight shell %}
+skydive client query "G.Flows().Has('Application', 'ICMPv4').Values('TrackingID')"
+[
+	"b53dafa2e10d19d5c1598e0bc723b7eb5bc3655a",
+	"b53dafa2e10d19d5c1598e0bc723b7eb5bc3655a"
+]
+{% endhighlight %}
+
+The `Gremlin` step `Dedup` de-duplicates flow according to the `TrackingID`.
+
+{% highlight shell %}
+skydive client query "G.Flows().Has('Application', 'ICMPv4').Dedup()"
+[
+	{
+		"UUID": "bad1803ce7caa70e7835356afdccacaa8e89b185",
+		"LayersPath": "Ethernet/IPv4/ICMPv4",
+		"Application": "ICMPv4",
+		"Link": {
+			"Protocol": "ETHERNET",
+			"A": "52:54:00:cc:e2:ef",
+			"B": "52:54:00:43:8c:a3",
+			"ID": 0
+		},
+		"Network": {
+			"Protocol": "IPV4",
+			"A": "192.168.121.112",
+			"B": "192.168.121.217",
+			"ID": 0
+		},
+		"ICMP": {
+			"Type": "ECHO",
+			"Code": 0,
+			"ID": 0
+		},
+		"Metric": {
+			"ABPackets": 5,
+			"ABBytes": 210,
+			"BAPackets": 5,
+			"BABytes": 210,
+			"Start": 1520545797027,
+			"Last": 1520545797027
+		},
+		"Start": 1520545797027,
+		"Last": 1520545797027,
+		"RTT": 208265,
+		"TrackingID": "b53dafa2e10d19d5c1598e0bc723b7eb5bc3655a",
+		"L3TrackingID": "52e47eab007caeb74cfba34f713242f939a11ce7",
+		"ParentUUID": "",
+		"NodeTID": "50037073-7862-5234-4996-e58cc067c69c",
+		"BNodeTID": "50037073-7862-5234-4996-e58cc067c69c",
+		"RawPacketsCaptured": 0
+	}
+]
+{% endhighlight %}
+
+We can also specify the capture from where we want to get the flow.
+
+{% highlight shell %}
+skydive client query "G.V('50037073-7862-5234-4996-e58cc067c69c').Flows().Has('Application', 'ICMPv4')"
+[
+	{
+		"UUID": "bad1803ce7caa70e7835356afdccacaa8e89b185",
+		"LayersPath": "Ethernet/IPv4/ICMPv4",
+		"Application": "ICMPv4",
+		"Link": {
+			"Protocol": "ETHERNET",
+			"A": "52:54:00:cc:e2:ef",
+			"B": "52:54:00:43:8c:a3",
+			"ID": 0
+		},
+		"Network": {
+			"Protocol": "IPV4",
+			"A": "192.168.121.112",
+			"B": "192.168.121.217",
+			"ID": 0
+		},
+		"ICMP": {
+			"Type": "ECHO",
+			"Code": 0,
+			"ID": 0
+		},
+		"Metric": {
+			"ABPackets": 5,
+			"ABBytes": 210,
+			"BAPackets": 5,
+			"BABytes": 210,
+			"Start": 1520545797027,
+			"Last": 1520545797027
+		},
+		"Start": 1520545797027,
+		"Last": 1520545797027,
+		"RTT": 208265,
+		"TrackingID": "b53dafa2e10d19d5c1598e0bc723b7eb5bc3655a",
+		"L3TrackingID": "52e47eab007caeb74cfba34f713242f939a11ce7",
+		"ParentUUID": "",
+		"NodeTID": "50037073-7862-5234-4996-e58cc067c69c",
+		"BNodeTID": "50037073-7862-5234-4996-e58cc067c69c",
+		"RawPacketsCaptured": 0
+	}
+]
+{% endhighlight %}
+
+It is also possible to start from a flow to get all the captured interfaces
+where this flow has been seen.
+
+```
+skydive client query "G.Flows().Has('Application', 'ICMPv4').CaptureNode().Values('ID')"
+[
+	"a5f2fd0d-1f30-46a9-75ff-6ba4c82426a1",
+	"99d511f5-5088-4c0d-67db-e45af33cb87c"
+]
+```
+
```

Comment:
you should add the query with TrackingID to show the same result

skydive client query "G.Flows().Has('TrackingID', 'b53dafa2e10d19d5c1598e0bc723b7eb5bc3655a').CaptureNode().Values('ID')"

After that it would be more clear for the reader, what TrackingID is

Reply:
done